apps/src/code-studio/components/header/ProjectImport.jsx

```diff
@@ -0,0 +1,124 @@
+/* globals dashboard, Craft */
+
+import $ from 'jquery';
+import React from 'react';
+
+import {convertBlocksXml} from '../../../craft/code-connection/utils';
+
+/**
+ * Show a popup dialog to collect an Hour of Code share link, and create a new
+ * channel-backed project from the associated LevelSource.
+ *
+ * Currently only supported for Minecraft Code Connection Projects and Minecraft
+ * Agent share links
+ */
+export default class ProjectImport extends React.Component {
+  importProject = () => {
+    if (!Craft) {
+      return;
```

Comment:
yuck! of course this yuckiness has to go somewhere and is a necessary evil for now given what you're doing, but I really don't like seeing it inside a react component. Would this seem any less gross if moved to headerRedux?

to be really clean, the Craft class would trigger a redux action which sets an importProject callback, so that all the Craft code can stay inside craft. but that seems like too much for now.

ok if you decide not to do anything here right now and def don't think we need to track this explicitly if we don't do it now, but when badness appears it can be a good prompt to at least identify a few steps along the path to making it better.

Reply:
Yeah, agreed that this is super gross. I don't think it's any more gross here than it was in the header code before, but it's definitely no better, either. I'll add a Jira card to track finding a better place/way to do this.

Reply:
100% agree that it's no more gross here than it was before, and that this is a good incremental step!